### PR TITLE
Documentation for Azure Storage

### PR DESCRIPTION
**Pull request by mehertz**

### CODE REVIEW

File: docs/mkdocs/docs/index.md [hunk 1]
```diff
@@ -3,9 +3,9 @@
 ## What is ArcticDB?
 
 ArcticDB is an embedded/serverless database engine designed to integrate with Pandas and the Python Data Science ecosystem. ArcticDB enables 
-you to store, retrieve and process DataFrames at scale, backed by commodity S3 storage.
+you to store, retrieve and process DataFrames at scale, backed by commodity object storage (S3-compatible storages and Azure Blob Storage are both supported).
```
Comment (qc00) [inline docs/mkdocs/docs/index.md:6]: Minor: `s/ are both supported//`

File: docs/mkdocs/docs/index.md [hunk 2]
```diff
@@ -31,12 +31,14 @@ pip install arcticdb
...
-ArcticDB is a storage engine designed for S3. As a result, you must have an available S3 bucket to store data using ArcticDB. 
+ArcticDB is a storage engine designed for object storage, but also supports local-disk storage using LMDB.
 
 !!! Storage Compatibility
 
     ArcticDB supports any S3 API compatible storage. It has been tested against AWS S3 and storage appliances like [VAST Universal Storage](https://vastdata.com/).
 
+    Beside S3, we also support Azure Blob Storage as well.
+
     ArcticDB also supports LMDB for local/file based storage - to use LMDB, pass an LMDB path as the URI: `Arctic('lmdb://path/to/desired/database')`.
```
Comment (qc00) [inline docs/mkdocs/docs/index.md:42]: This line is now redundant.

File: python/arcticdb/arctic.py [hunk 3]
```diff
@@ -38,40 +38,88 @@ def __init__(self, uri: str, encoding_version: EncodingVersion = EncodingVersion
...
+                Please refer to https://github.com/Azure/azure-sdk-for-cpp/blob/24ed290815d8f9dbcd758a60fdc5b6b9205f74e0/sdk/core/azure-core/inc/azure/core/http/http_status_code.hpp for
+                more details of provided status codes. Note that due to a bug in Azure C++ SDK (https://github.com/Azure/azure-sdk-for-cpp/issues/4738), Azure may not give meaningful status codes and
+                reason phrases in the exception. To debug these instances, please set the environment variable ``export AZURE_LOG_LEVEL`` to ``1`` to turn on the SDK debug logging.
+
+
+            LMDB
+            ----
+
+                The LMDB URI connection scheme has the form ``lmdb:///<path to store LMDB files>``. There are no options
+                available for the LMDB URI connection scheme.
```
Comment (qc00) [inline python/arcticdb/arctic.py:122]: The length of this now makes it impractical to view in an IDE and an annoyance if the stack trace included this function in Pytest (which may dump the full source).

It's more sensible to move this in a docs page now.

Comment (mehertz) [reply python/arcticdb/arctic.py:122]: I'll change the line in this PR - would rather move in another PR as this PR is blocking the next release.

Comment (mehertz) [reply python/arcticdb/arctic.py:122]: https://github.com/man-group/ArcticDB/issues/565

**Review by GemDot — APPROVED** [review]

**Review by qc00 — APPROVED** [review]

**Pull request merged**